strings/strip_url_params.py

```diff
@@ -0,0 +1,106 @@
+            key_value_string = []
+            string = ''
+            for char in query_string:
+                if char.isdigit():
+                    key_value_string.append(string + char)
+                    string = ''
+                else:
+                    string += char
+            dict = defaultdict(int)
+            # logic for checking whether we should add the string to our result
+            for i in key_value_string:
+                _token = i.split('=')
+                if _token[0]:
+                    length = len(_token[0])
+                    if length == 1:
+                        if _token and (not(_token[0] in dict)):
+                            if params_to_strip:
+                                if _token[0] != params_to_strip[0]:
+                                    dict[_token[0]] = _token[1]
+                                    result = result + _token[0] + '=' + _token[1]
+                            else:
+                                if not _token[0] in dict:
+                                    dict[_token[0]] = _token[1]
+                                    result = result + _token[0] + '=' + _token[1]
+                    else:
+                        check = _token[0]
+                        letter = check[1]
+                        if _token and (not(letter in dict)):
+                            if params_to_strip:
+                                if letter != params_to_strip[0]:
+                                    dict[letter] = _token[1]
+                                    result = result + _token[0] + '=' + _token[1]
+                            else:
+                                if not letter in dict:
+                                    dict[letter] = _token[1]
+                                    result = result + _token[0] + '=' + _token[1]
+    return result
+
+# A very friendly pythonic solution (easy to follow)
+def strip_url_params(url, param_to_strip=[]):
+    if '?' not in url:
+        return url
+
+    queries = (url.split('?')[1]).split('&')
+    queries_obj = [query[0] for query in queries]
+    for i in range(len(queries_obj) - 1, 0, -1):
+        if queries_obj[i] in param_to_strip or queries_obj[i] in queries_obj[0:i]:
+            queries.pop(i)
+
+    return url.split('?')[0] + '?' + '&'.join(queries)
+
+
+# Here is my friend's solution using python's builtin libraries
+import urlparse
+import urllib
+def strip_url_params(url, strip=None):
+    if not strip: strip = []
+    
+    parse = urlparse.urlparse(url)
+    query = urlparse.parse_qs(parse.query)
+    
+    query = {k: v[0] for k, v in query.iteritems() if k not in strip}
+    query = urllib.urlencode(query)
+    new = parse._replace(query=query)
+    
+    return new.geturl()
+
+
+import unittest
+class TestSuite(unittest.TestCase):
+
+    def test_valid(self):
+        self.assertEqual(strip_url_params("www.saadbenn.com?a=1&b=2&a=2"), "www.saadbenn.com?a=1&b=2")
+        
+    def test_invalid(self):
```

Comment:
We don't write tests for cases that are invalid! The tests will fail in such case, Travis will also show error.